### DISCUSSION TITLE
use Z3 in intellij ubuntu

### DISCUSSION THREAD

**Discussion by maksglukhov** [Z3Prover/z3#5583] [Q&A]
Hello, 

I'm new to use Z3 in my project. I'm using Intellij on ubuntu 20.04, 64-bit.
I cannot install Z3 in my Intellij. I have added the .jar in libraries of my project and i have done **"import com.microsoft.z3.*;"**
But i have the folowing error:
**no libz3java in java.library.path: [/usr/java/packages/lib, /usr/lib64, /lib64, /lib, /usr/lib]**
Then i have added VM option : 
**Djava.library.path="~z3/bin"**

But then i have the error :
**Error: Could not find or load main class Djava.library.path=~z3.bin
Caused by: java.lang.ClassNotFoundException: Djava.library.path=~z3.bin**

I also tried
**Djava.library.path="~z3/bin/libz3java.so"**
but it's not working any more.

Thank you for your help!

**Comment by wintersteiger** [comment]
It thinks that `Djava.library.path="~z3/bin"` is a class name and it removes the slashes from it. You want a `-D`, but I don't know what the IntelliJ GUI expects. It may have a dedicated entry in the project properties for just this option. This might help: https://stackoverflow.com/questions/19308010/how-to-set-the-java-library-path-in-intellij-idea

**Reply by maksglukhov** [reply]
ty, but i have already saw this page and it didn't help me...

**Comment by wintersteiger** [comment]
Did you try adding a dash to your `D`?
